Utilities/GetPSGalleryItemsForAuthor.ps1

```diff
@@ -0,0 +1,100 @@
+﻿
+<#PSScriptInfo
+
+.VERSION 0.1.0
+
+.GUID 788b94eb-f6f0-400d-b259-2e15c3f102ee
+
+.AUTHOR J. Keith Bankston
+
+.COMPANYNAME Microsoft
+
+.COPYRIGHT Microsoft, 2018
+
+.TAGS GalleryUtility,GDPR
+
+.LICENSEURI 
+
+.PROJECTURI 
+
+.ICONURI 
+
+.RELEASENOTES
+  Version 0.1.0 - Initial Version. See description for use information. 
+
+
+.PRIVATEDATA 
+
+#>
+
+<# 
+
+.DESCRIPTION 
+This script will save all versions of all scripts and modules published to the PowerShell Gallery by a given Author.
+The Author is specified in the manifest for each item, and is expected to be consistent for a given user.
+
+The script assumes that there is at least 1 version of every item that is listed. 
+This can be verified by logging into the PowerShell Gallery, navigating to https://www.powershellgallery.com/account/Packages.
+If there are items that are hidden, they will be shown at the bottom of that page under the header: "Unlisted Items"
+
+Use instructions: 
+* Provide the AuthorName, which is the Author used in the item manifest. 
+  Note that the Author is not guaranteed to match the Owner, which is the account name for those who publish the item.
+* Provide the LocalFolder to place all the PowerShell Gallery items into. 
+
+For questions on this script, email cgadmin@microsoft.com 
+
+#> 
+
+[CmdletBinding()]
+param(
+    [Parameter(Mandatory = $true)]
+    [string] $AuthorName,
+
+    [Parameter(Mandatory = $true)]
+    [string] $LocalFolder
+)
+
+
+If (-not (test-path -path $LocalFolder)) {
+   write-verbose "Creating folder $LocalFolder"
+   new-item $LocalFolder -itemtype Directory
+}
+    
+
+Write-Host "Retrieving all modules published by the author $AuthorName"
+
+find-module -repository psgallery |Where-Object { 
+        if($_.Author.count -eq 1) {
+            $_.Author -eq $AuthorName 
+        }
+        else {
+            $_.Author -contains $AuthorName
+        }
+    } | ForEach-Object {
+            Write-Verbose "Retrieving all versions of $($_.Name)"
+
+            Find-Module -Name $_.Name -repository psgallery -AllowPrerelease -AllVersions |  Foreach-Object { 
+                Save-Module -Name $_.Name -RequiredVersion $_.Version -Path $LocalFolder -Repository psgallery -AllowPrerelease
```

Comment:
Please consider specifying -AcceptLicense here and also in Save-Script usage below.

Reply:
Worth considering, but I disagree. This can be used for more than GDPR purposes, it can also be used to acquire everything published by VMWare. In that case, I want the user to make the conscious choice, and not do it for them in a way they would not see. We could add it as a parameter, I suppose, if you think it's worth it.